js-library/src/services/ens.ts

```diff
@@ -0,0 +1,176 @@
+import { utils, Contract, Signer, providers } from 'ethers'
+import { NULL_ADDRESS } from '../constants/constants'
+import ENSRegistryContract from '../contracts/ENSRegistry.sol/ENSRegistry.json'
+import PublicResolverContract from '../contracts/PublicResolver.sol/PublicResolver.json'
+import SubdomainRegistrarContract from '../contracts/SubdomainRegistrar.sol/SubdomainRegistrar.json'
+import { waitTransaction } from '../utils/tx'
+import { joinPublicKey, isPublicKeyValid, splitPublicKey } from '../utils/keys'
+
+const { keccak256, toUtf8Bytes, namehash } = utils
+
+/**
+ * Contract addresses
+ */
+export const ENS_REGISTRY_ADDRESS = process.env.ENS_REGISTRY_ADDRESS
+export const PUBLIC_RESOLVER_ADDRESS = process.env.PUBLIC_RESOLVER_ADDRESS
+export const SUBDOMAIN_REGISTRAR_ADDRESS = process.env.SUBDOMAIN_REGISTRAR_ADDRESS
+
+/**
+ * Contract metadata
+ */
+export const ENSRegistryMeta = ENSRegistryContract
+export const PublicResolverMeta = PublicResolverContract
+export const SubdomainRegistrarMeta = SubdomainRegistrarContract
+
+/**
+ * ENS Class
+ * Provides interface for interaction with the ENS smart contracts
+ */
+export class ENS {
+  private _provider: providers.JsonRpcProvider
+  private ensRegistryContract: Contract
+  private subdomainRegistrarContract: Contract
+  private publicResolverContract: Contract
+
+  constructor(rpcUrl = 'http://127.0.0.1:9545/', private domain = 'fds.eth') {
+    this._provider = new providers.JsonRpcProvider(rpcUrl)
+
+    this.ensRegistryContract = new Contract(
+      ENS_REGISTRY_ADDRESS as string,
+      ENSRegistryContract.abi,
+      this._provider,
+    )
+    this.publicResolverContract = new Contract(
+      PUBLIC_RESOLVER_ADDRESS as string,
+      PublicResolverContract.abi,
+      this._provider,
+    )
+    this.subdomainRegistrarContract = new Contract(
+      SUBDOMAIN_REGISTRAR_ADDRESS as string,
+      SubdomainRegistrarContract.abi,
+      this._provider,
+    )
+  }
+
+  /**
+   * returns RPC provider
+   */
+  public get provider(): providers.JsonRpcProvider {
+    return this._provider
+  }
+
+  /**
+   * Connects signer to the smart contracts
+   * @param signerOrProvider An instance of ethers.js Wallet or any other signer
+   */
+  public connect(signerOrProvider: string | providers.Provider | Signer) {
```

Comment:
I would call this function, if the `signerOrProvider` optional parameter has been passed in the constructor.